### PR TITLE
udpt: 2017-09-27 -> 3.1.1

### PR DESCRIPTION
**Pull request by makefu**
###### Motivation for this change
udpt did a complete rewrite from cpp to rust.
This Commit uses rust with cargo to build the package.
The Configuration format changed form .conf to .toml and
the binary changed from udpt to udpt-rs

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [NA] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date - update release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2009.xml [hunk 1]
```diff
@@ -525,6 +525,14 @@ systemd.services.nginx.serviceConfig.ReadWritePaths = [ "/var/www" ];
      automatically if <literal>stateVersion</literal> is 20.09 or higher.
     </para>
    </listitem>
+   <listitem>
+    <para>
+      <literal>udpt</literal> experienced a complete rewrite from C++ to rust. The configuration format changed from ini to toml.
+      The new configuration documentation can be found at
+      <link xlink:href="https://naim94a.github.io/udpt/config.html">the official website</link> and example
+      configuration is packaged in <literal>${udpt}/usr/lib/udpt/udpt.toml</literal>.
```
Comment (flokli) [inline nixos/doc/manual/release-notes/rl-2009.xml:533]:
```suggestion
      configuration is packaged in <literal>${udpt}/lib/udpt/udpt.toml</literal>.
```

File: pkgs/servers/udpt/default.nix [hunk 2]
```diff
@@ -1,45 +1,21 @@
...
+  verifyCargoDeps = true;
 
-  installPhase = ''
-    mkdir -p $out/bin $out/etc/
-    cp udpt $out/bin
-    cp ../udpt.conf $out/etc/
-    # without this, the resulting binary is unstripped.
-    runHook postInstall
+  postInstall = ''
+    install -D udpt.toml $out/usr/lib/udpt/udpt.toml
```
Comment (flokli) [inline pkgs/servers/udpt/default.nix:18]:
```suggestion
    install -D udpt.toml $out/lib/udpt/udpt.toml
```

**Review by flokli — COMMENTED** [review]
We usually drop `/usr` from the locations.

And generally, if this is an example config, maybe `$out/etc/udpt/udpt.toml` or `$out/share/udpt/udpt.toml` are more appropriate?

File: nixos/doc/manual/release-notes/rl-2009.xml (hunk 1 above)
Comment (makefu) [reply nixos/doc/manual/release-notes/rl-2009.xml:533]: I agree, is there a convention which files go where? Upstream uses `/usr/src` which didn't really convince me as well :)

**Comment by makefu** [comment]
i went for `$out/share/udpt/udpt.toml`

**Review by flokli — CHANGES_REQUESTED** [review]
I get hash mismatches in the fixed-output derivation both on ofborg and locally.

**Comment by makefu** [comment]
@flokli okay this is weird, i get a different vendor checksum on the current nixpkgs master compared to nixos-20.03 ... were there some breaking changes on master regarding cargo checksum generation?

**Comment by flokli** [comment]
https://github.com/NixOS/nixpkgs/issues/79975 probably.

**Comment by makefu** [comment]
@flokli thanks for investigating! i've fixed the cargo hash.
Cheers!

**Pull request merged**